drape/drape_tests/CMakeLists.txt

```diff
@@ -36,11 +41,11 @@ if (WITH_SYSTEM_PROVIDED_3PARTY)
   target_link_libraries(${PROJECT_NAME} GTest::gtest)
 else()
   add_subdirectory("${CMAKE_CURRENT_LIST_DIR}/../../3party/googletest" "${CMAKE_CURRENT_BINARY_DIR}/googletest")
+  target_include_directories(${PROJECT_NAME} PUBLIC ${OMIM_ROOT}/3party/glm)
 endif()
 
 target_link_libraries(${PROJECT_NAME}
   qt_tstfrm
-  indexer  # For StyleReader in static_texture_tests
-  drape
```

Comment:
Why is this change and source list changes needed? It looks unnatural and complex for cmake.

Reply:
If you want to run all the tests, including those that rely on mocks in `drape_tests/gl_functions.cpp`, you can't link `drape` because, that would violate the one definition rule. Hence we don't link drape here, and also change to sources to not inlcude the original  `drape/gl_functions.cpp` but the mock `drape_tests/gl_functions.cpp` instead.

Reply:
Thanks for the explanation! Would it be clearer to use #ifdef in the mentioned files?

Reply:
Can test's cmake file remove/replace a source from another target library